packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -50,27 +45,30 @@ interface TerminalCSSProperties {
 }
 
 @injectable()
-export class TerminalWidget extends BaseWidget implements StatefulWidget {
+export class TerminalWidgetImpl extends BaseWidget implements TerminalWidget, StatefulWidget {
 
     private terminalId: number | undefined;
     private term: Xterm.Terminal;
+    private readonly TERMINAL = "Terminal";
+    private readonly onTermDidClose = new Emitter<TerminalWidget>();
     protected restored = false;
     protected closeOnDispose = true;
+    protected waitForConnection: Deferred<MessageConnection>;
```

Comment:
`Deferred<MessageConnection> | undefined` - make sure that the type reflects possible values